hazelcast/src/hazelcast/util/impl/Executor.cpp

```diff
@@ -143,9 +134,22 @@ namespace hazelcast {
 
             std::string SimpleExecutorService::Worker::generateThreadName(const std::string &prefix) {
                 std::ostringstream out;
-                out << prefix << "-" << (++THREAD_ID_GENERATOR);
+                out << prefix << (++executorService.threadIdGenerator);
                 return out.str();
             }
+
+            void SimpleExecutorService::Worker::shutdown() {
+                workQueue.clear();
```

Comment:
ExecutorService shutdown should not clear the queue. It should wait all tasks to finish and after this is called worker should not accept new tasks.